tools/clang/lib/Sema/SemaHLSL.cpp

```diff
@@ -13984,7 +13984,14 @@ bool Sema::DiagnoseHLSLDecl(Declarator &D, DeclContext *DC, Expr *BitWidth,
     QualType eltQt(qt->getArrayElementTypeNoTypeQual(), 0);
     while (eltQt->isArrayType())
       eltQt = QualType(eltQt->getArrayElementTypeNoTypeQual(), 0);
-
+    if (hlslSource->IsWaveMatrixType(eltQt)) {
+      std::string typeName(
+          g_ArBasicTypeNames[hlslSource->GetTypeElementKind(eltQt)]);
+      const int waveMatrixDiagSelect = 1;
+      Diag(D.getLocStart(), diag::err_hlsl_array_disallowed)
+          << typeName << waveMatrixDiagSelect;
```

Comment:
Putting this in a const int is harder to read. The common pattern in Clang is to use an inline comment:
```suggestion
          << typeName << \* declaration *\ 1;
```

Reply:
Updated.